bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -144,8 +144,17 @@ private String getFromApi(String uri) throws MeaterException, MeaterAuthenticati
 
                     if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
                         // This will currently not happen because "WWW-Authenticate" header is missing; see below.
-                        logger.debug("getFromApi failed, HTTP status: 401");
-                        login();
+                        logger.debug("getFromApi failed, authentication failed, HTTP status: 401");
+                        if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                            /*
+                             * When contextId is not valid, the service will respond with HTTP code 401 without
+                             * any "WWW-Authenticate" header, violating RFC 7235. Jetty will then throw
+                             * HttpResponseException. We need to handle this in order to attempt
+                             * reauthentication.
+                             */
+                            logger.debug("getFromApi failed, authentication failed, HTTP status: 401");
```

Comment:
This is already checked and logged above, so these lines can be omitted. Only request was to throw `MeaterAuthenticationException` instead of calling login() on our own.

Reply:
Fixed!

Reply:
Sorry, but all commented lines should be removed. :-) Status code is checked twice, and the multi-line comment is only relevant for the work-around later in the method.

Reply:
Fixed!